identity/src/lib.rs

```diff
@@ -98,34 +97,71 @@ impl zeroize::Zeroize for proto::PrivateKey {
 ))]
 impl From<&PublicKey> for proto::PublicKey {
     fn from(key: &PublicKey) -> Self {
+        // #[allow(deprecated)]
+        // match key {
+        //     #[cfg(feature = "ed25519")]
+        //     PublicKey::Ed25519(key) => proto::PublicKey {
+        //         Type: proto::KeyType::Ed25519,
+        //         Data: key.encode().to_vec(),
+        //     },
+        //     #[cfg(all(feature = "rsa", not(target_arch = "wasm32")))]
+        //     PublicKey::Rsa(key) => proto::PublicKey {
+        //         Type: proto::KeyType::RSA,
+        //         Data: key.encode_x509(),
+        //     },
+        //     #[cfg(feature = "secp256k1")]
+        //     PublicKey::Secp256k1(key) => proto::PublicKey {
+        //         Type: proto::KeyType::Secp256k1,
+        //         Data: key.encode().to_vec(),
+        //     },
+        //     #[cfg(feature = "ecdsa")]
+        //     PublicKey::Ecdsa(key) => proto::PublicKey {
+        //         Type: proto::KeyType::ECDSA,
+        //         Data: key.encode_der(),
+        //     },
+        // }
+
         #[allow(deprecated)]
-        match key {
+        #[cfg(any(
+            feature = "ed25519",
+            all(feature = "rsa", not(target_arch = "wasm32")),
+            feature = "secp256k1",
+            feature = "ecdsa"
+        ))]
+        match (
+            key.clone().try_into_ed25519(),
+            key.clone().try_into_rsa(),
+            key.clone().try_into_secp256k1(),
+            key.clone().try_into_ecdsa(),
+        ) {
```

Comment:
@thomaseizinger How do you prefer to handle this match: in fact it is a little complicated with lots of `cfg` conditions, for the moment it does not work but in the idea how do you see this implementation?

Reply:
Why do we have to move away from the original code? We can make the field `pub(crate)` and just match on the inner public key, provided that we use the correct `cfg` on each branch as before.

Reply:
@thomaseizinger I've just pushed a new version where:
- I set `pub(crate) publickey: PublicKeyInner,` inside `PublicKey`.
- I set `pub(crate) enum PublicKeyInner` to have access to items such as `PublicKeyInner::Ed25519` in `lib.rs`